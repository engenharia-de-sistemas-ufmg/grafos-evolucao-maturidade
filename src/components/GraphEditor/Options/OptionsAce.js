import React, { useState } from 'react';
import PropTypes from 'prop-types';
import AceEditor from 'react-ace';
import { optionsType } from '../../../utils/types';
import 'brace/mode/json';
import 'brace/theme/textmate';
import { isValidJson, fromAceEditor, toAceEditor } from '../../../utils/json';
import * as labels from '../../../utils/labels';

const OptionsAce = ({ lang, options, update }) => {
  const [tmpOptions, setTmpOptions] = useState(toAceEditor(options));
  const [isValidOptions, validateOptions] = useState(true);
  return (
    <div>
      <button
        type="button"
        className="button is-primary is-small"
        style={{ marginBottom: '12px' }}
        disabled={!isValidOptions}
        onClick={() => {
          update(fromAceEditor(tmpOptions));
        }}
      >
        {labels.update[lang]}
      </button>
      <AceEditor
        value={tmpOptions}
        mode="json"
        theme="textmate"
        name="OPTIONS_EDITOR"
        tabSize={2}
        onChange={(value) => {
          setTmpOptions(value);
          validateOptions(isValidJson(value));
        }}
        editorProps={{
          $blockScrolling: Infinity,
        }}
      />
    </div>
  );
};

OptionsAce.propTypes = {
  lang: PropTypes.string.isRequired,
  options: optionsType.isRequired,
  update: PropTypes.func.isRequired,
};

export default OptionsAce;
