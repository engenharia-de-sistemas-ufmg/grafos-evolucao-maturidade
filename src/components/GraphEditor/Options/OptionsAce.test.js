import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import OptionsAce from './OptionsAce';

describe('OptionsAce', () => {
  Enzyme.configure({ adapter: new Adapter() });

  test('should match snapshot', () => {
    const component = shallow(
      <OptionsAce
        lang="pt"
        options={{ height: '500px' }}
        update={() => { }}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
