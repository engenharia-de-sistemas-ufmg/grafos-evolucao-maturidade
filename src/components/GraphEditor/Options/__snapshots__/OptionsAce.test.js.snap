// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsAce should match snapshot 1`] = `
<div>
  <button
    className="button is-small is-text"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "float": "right",
      }
    }
    type="button"
  >
    Atualizar
  </button>
  <ReactAce
    cursorStart={1}
    editorProps={
      Object {
        "$blockScrolling": Infinity,
      }
    }
    enableBasicAutocompletion={false}
    enableLiveAutocompletion={false}
    enableSnippets={false}
    focus={false}
    fontSize={12}
    height="500px"
    highlightActiveLine={true}
    maxLines={null}
    minLines={null}
    mode="json"
    name="OPTIONS_EDITOR"
    navigateToFileEnd={true}
    onChange={[Function]}
    onLoad={null}
    onPaste={null}
    onScroll={null}
    placeholder={null}
    readOnly={false}
    scrollMargin={
      Array [
        0,
        0,
        0,
        0,
      ]
    }
    setOptions={Object {}}
    showGutter={true}
    showPrintMargin={true}
    style={Object {}}
    tabSize={2}
    theme="textmate"
    value="{
 \\"height\\": \\"500px\\"
}"
    width="100%"
    wrapEnabled={false}
  />
</div>
`;
